pkg/gardenlet/bootstrap/bootstrap.go

```diff
@@ -43,160 +34,56 @@ import (
-// certificates) and send it to API server, then it will watch the object's
-// status, once approved by API server, it will return the API server's issued
-// certificate (pem-encoded). If there is any errors, or the watch timeouts, it
-// will return an error.
-func RequestCertificate(ctx context.Context, certificateClient certificatesv1beta1client.CertificateSigningRequestInterface, privateKeyData []byte, commonName string, organization []string) (certData []byte, csrName string, err error) {
-	subject := &pkix.Name{
-		Organization: organization,
-		CommonName:   commonName,
-	}
-
-	privateKey, err := keyutil.ParsePrivateKeyPEM(privateKeyData)
+// CreateBootstrapClientFromKubeconfig creates a CertificatesV1beta1Client client from a given kubeconfig
+func CreateBootstrapClientFromKubeconfig(kubeconfig []byte) (*certificatesv1beta1client.CertificatesV1beta1Client, *rest.Config, error) {
+	bootstrapClientConfig, err := clientcmd.NewClientConfigFromBytes(kubeconfig)
 	if err != nil {
-		return nil, "", fmt.Errorf("invalid private key for certificate request: %v", err)
-	}
-	csrData, err := certutil.MakeCSR(privateKey, subject, nil, nil)
-	if err != nil {
-		return nil, "", fmt.Errorf("unable to generate certificate request: %v", err)
-	}
-
-	usages := []certificatesv1beta1.KeyUsage{
-		certificatesv1beta1.UsageDigitalSignature,
-		certificatesv1beta1.UsageKeyEncipherment,
-		certificatesv1beta1.UsageClientAuth,
+		return nil, nil, err
 	}
 
-	// The Signer interface contains the Public() method to get the public key.
-	signer, ok := privateKey.(crypto.Signer)
-	if !ok {
-		return nil, "", fmt.Errorf("private key does not implement crypto.Signer")
-	}
-
-	name, err := digestedName(signer.Public(), subject, usages)
+	bootstrapConfig, err := bootstrapClientConfig.ClientConfig()
 	if err != nil {
-		return nil, "", err
+		return nil, nil, err
 	}
-
-	req, err := csr.RequestCertificate(certificateClient, csrData, name, usages, privateKey)
+	bootstrapClient, err := certificatesv1beta1client.NewForConfig(bootstrapConfig)
 	if err != nil {
-		return nil, "", err
+		return nil, nil, err
 	}
+	return bootstrapClient, bootstrapConfig, nil
+}
 
-	childCtx, cancel := context.WithTimeout(ctx, 3600*time.Second)
-	defer cancel()
-
-	certData, err = csr.WaitForCertificate(childCtx, certificateClient, req)
+// RequestBootstrapKubeconfig creates a kubeconfig with a signed certificate using the given bootstrap client
+// returns the kubeconfig []byte representation, the CSR name, the seed name or an error
+func RequestBootstrapKubeconfig(ctx context.Context, logger logrus.FieldLogger, seedClient client.Client, bootstrapClient certificatesv1beta1client.CertificatesV1beta1Interface, bootstrapConfig *rest.Config, gardenClientConnection *config.GardenClientConnection, seedName, bootstrapTargetCluster string) ([]byte, string, string, error) {
+	certData, privateKeyData, csrName, err := certificate.RequestCertificate(ctx, logger, bootstrapClient, seedName, nil)
```

Comment:
Pass proper subject here instead of `nil`. See my other comment in `pkg/gardenlet/bootstrap/certificate/certificate.go`.